dollarskip.c

```diff
@@ -12,14 +15,22 @@ int main(int argc, char **argv)
 			for (int i=1; i<argc; i++){
 				strcat(strcat(command,argv[i])," ");
 			} //add all the command (cmd args) to command
+			/****
+    			* remove the space from the end of the command
+    			* for example the command is "ls -l", it takes up 5 characters from the array (command variable) + a space at the end caused by the for loop above + a character at the end that is reserved for a NUL character.
+    			* in this case 'strlen(command)' will return 6, remove 1, and you get the location of the character in the array containing the extra space.
+    			* replace the space with NUL (null terminator), the end of the string.
+   	 		****/
 			command[strlen(command) - 1] = '\0';
 			strcat(command, "\'");// add "'"
-		} else {
-			for (int i=1; i<argc; i++){
+		} else { //if the SHELL env var doesn't exist or contain anything,
+			for (int i=1; i<argc; i++){ //put the command in 'command',
 				strcat(strcat(command,argv[i])," ");
 			}
-			command[strlen(command) - 1] = '\0';
+			command[strlen(command) - 1] = '\0'; //and remove the space at the end.
```

Comment:
```suggestion
			command[strlen(command) - 1] = '\0'; // and remove the NULL space  at the end.
```

Reply:
Wait, does this remove the null space. Looks like it add ones after removing the ` ` space

Reply:
since the last character should be a NULL (`\0`) I think its actually replacing the NULL with a NULL, so doing nothing.